lib/mixlib/install/backend/package_router.rb

```diff
@@ -0,0 +1,222 @@
+          # artifactory.
+
+          if options.channel == :unstable
+            # We check if "artifacts" field contains something since it is only
+            # populated with the build record if "artifact.module.build" exists.
+            items.reject! { |i| i["artifacts"].nil? }
+          end
+
+          # We are only including a single property, version and that exists
+          # under the properties in the following structure:
+          # "properties" => [ {"key"=>"omnibus.version", "value"=>"12.13.3"} ]
+          items.map! { |i| i["properties"].first["value"] }
+          items.uniq
+        end
+
+        #
+        # Get artifacts for the latest version, channel and product_name
+        #
+        # @return [Array<ArtifactInfo>] Array of info about found artifacts
+        def latest_version
+          # Get the list of builds from the REST api.
+          # We do this because a user in the readers group does not have
+          # permissions to run aql against builds.
+          builds = get("/api/build/#{omnibus_project}")
+
+          # Output we get is something like:
+          # {
+          #   "buildsNumbers": [
+          #     {"uri"=>"/12.5.1+20151213083009", "started"=>"2015-12-13T08:40:19.238+0000"},
+          #     {"uri"=>"/12.6.0+20160111212038", "started"=>"2016-01-12T00:25:35.762+0000"},
+          #     ...
+          #   ]
+          # }
+          # First we sort based on started
+          builds["buildsNumbers"].sort_by! { |b| b["started"] }.reverse!
+
+          # Now check if the build is in the requested channel or not
+          builds["buildsNumbers"].each do |build|
+            version = build["uri"].delete("/")
+            artifacts = artifacts_for_version(version)
+
+            return artifacts unless artifacts.empty?
+          end
+
+          # we could not find any matching artifacts
+          []
+        end
+
+        #
+        # Get artifacts for a given version, channel and product_name
+        #
+        # @return [Array<ArtifactInfo>] Array of info about found artifacts
+        def artifacts_for_version(version)
+          results = get("/api/#{options.channel}/#{omnibus_project}/#{version}/artifacts")["results"]
+
+          # Merge properties to a flat Hash
+          results.collect! do |result|
+            {
+              "filename" => result["name"],
+            }.merge(
+              map_properties(result["properties"])
+            )
+          end
+
+          # Convert results to build records
+          results.map { |a| create_artifact(a) }
+        end
+
+        #
+        # GET request
+        #
+        def get(url)
+          uri = URI.parse(endpoint)
+          http = Net::HTTP.new(uri.host, uri.port)
+          http.use_ssl = (uri.scheme == "https")
+
+          full_path = File.join(uri.path, url)
+          request = Net::HTTP::Get.new(full_path)
+
+          res = http.request(request)
+
+          body = if res.code != "200"
```

Comment:
I really need to change this as this doesn't apply to `get("/api/build/#{omnibus_project}")` in `latest_version`